packages/atlas/src/providers/joystream/provider.tsx

```diff
@@ -147,8 +147,10 @@ const useJoystreamUtilFns = (joystream: Remote<JoystreamLib> | undefined, proxyC
   const getCurrentBlockMsTimestamp = useCallback(() => currentBlockMsTimestampRef.current, [])
 
   return {
+    tokenPrice: tokenPriceRef.current,
```

Comment:
There's a slight problem with this approach: The context will not be updated when this value changes, i.e. changes won't trigger a rerender across users of this value. I think the approach with getter functions is safer since you are always guaranteed to get latest value when you trigger the function

Reply:
Ah but there's problem with my approach as well, the price can be not fetched yet when the component renders so it will get the old value. But current code has the same problem, what can we do to solve that? I guess we could keep the price inside a state instead of a ref?

Reply:
On some refreshes, I will get constant `$0` in the widget